test/withdraw-test.ts

```diff
@@ -511,7 +511,7 @@ describe('withdraw', function () {
       // in callback, EVIL token calls transferFrom(alice.address, bob.address, 1e6)
       await expect(
         comet.connect(alice).withdraw(EVIL.address, 1e6)
-      ).to.be.revertedWith("custom error 'NotCollateralized()'");
+      ).to.be.revertedWith("custom error 'TransferOutFailed()'");
```

Comment:
Now the accounting in comet won't be mistaken by re-entrancy withdraw reaching NotColalteralized() error.
Instead, it will be blocked and throw TransferInFailed() error right away.

Reply:
Shouldn't it throw the re-entrancy error instead?

Reply:
Yeah it was bit weird,  but since the error is thrown in the nested transfer() function, so it gets sort of "re-wrap" into TransferInFailed() as the code here doesn't re-raise the error (or should we make it re-raise the error?) 
```
uint256 preTransferBalance = ERC20(asset).balanceOf(address(this));
        (bool success, bytes memory returndata) = asset.call(abi.encodeWithSelector(ERC20.transferFrom.selector, from, address(this), amount));
        if (!success || !(returndata.length == 0 || abi.decode(returndata, (bool)))) {
            revert TransferInFailed();
        }
        return ERC20(asset).balanceOf(address(this)) - preTransferBalance;
```

From code we can see now the transferFrom revert won't reveirt right away with returndata, but instead when it fails, we throw `TransferInFailed()`.